### PR TITLE
Implement bottom navigation for mobile devices

### LINKED ISSUE #1420: Bottom navigation
Zrzut ekranu 2021-09-28 o 14 13 32
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=1341%3A94962

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #1420 
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=1341%3A94962

### CODE REVIEW

File: src/components/BottomNav/BottomNav.style.ts
Comment (kdembler) [inline src/components/BottomNav/BottomNav.style.ts]: Let's use `transform: translateY()` instead, it has much better transition performance than `bottom`

File: src/components/BottomNav/BottomNav.style.ts
Comment (kdembler) [inline src/components/BottomNav/BottomNav.style.ts]: With `translate` you can just use `-100%`

File: src/components/BottomNav/BottomNav.tsx
Comment (kdembler) [inline src/components/BottomNav/BottomNav.tsx]: What's the point of this?

File: src/config/nav.tsx
Comment (kdembler) [inline src/config/nav.tsx]: Let's keep the prefix 🙏  e.g. `viewerNavItems`

**Review by kdembler — CHANGES_REQUESTED** [review]
I think the UX could be improved a bit:
1) First off, I think we may have the scroll direction the other way around. I'd expect that when I open the app and start scrolling down (the page content goes up), the navigation disappears. As long as I go down and scroll further content it remains hidden. Only if I go back and scroll up, it appears back. I'm pretty sure that's what Adam had in mind as well, although I see that docs in Figma can be kind of misleading.
2) Another thing, I think the navigation should be visible right after you open the app on mobile and disappear only once you start scrolling. Also, when I navigate to another page it hides after the navigation. I believe it should be visible (for example to let the user navigate further) and only hide once scrolled.

File: src/components/BottomNav/BottomNav.style.ts
Comment (drillprop) [inline src/components/BottomNav/BottomNav.style.ts]: I think we could remove this block and use `align-items: center` instead

File: src/components/BottomNav/BottomNav.tsx
Comment (drillprop) [inline src/components/BottomNav/BottomNav.tsx]: Font is different in the design: 
image 
You should use our `Text` with `button3 ` variant here.

But... on the other hand, we could stick to `Inter` here, because it looks weird when I applied `PxGrotesk`. Let's ask designer about this, maybe?

Comment (drillprop) [reply src/components/BottomNav/BottomNav.tsx]: Ok, talked to Adam.  You should change this to `PxGrotesk`, also you should update the font in desktop navigation( you'll need to update `::after` pseudo-element probably)

Comment (toiletgranny) [reply src/components/BottomNav/BottomNav.tsx]: @drillprop knowing that soon we'll be replacing `PxGrotesk` with a completely different typeface altogether, I'd suggest sticking with `PxGrotesk` for now, to keep the implementation and design consistent when this happens.

**Comment by toiletgranny** [comment]
100% agree with what @kdembler said.

> 1. First off, I think we may have the scroll direction the other way around.

That's 100% correct: the bottom navigation bar should hide upon scrolling downwards and reappear upon scrolling upwards. Sorry if the Figma documentation was misleading. 🙈 

> 2. Another thing, I think the navigation should be visible right after you open the app on mobile and disappear only once you start scrolling. Also, when I navigate to another page it hides after the navigation. I believe it should be visible (for example to let the user navigate further) and only hide once scrolled.

100% agree with this. Ideally, the bottom navigation bar is visible by default and gets hidden upon scrolling down.

Other than this, now that I've played with it myself I'm not sure if the `128px` threshold is not too generous. Perhaps we could try with something like this:
1.  Upon scrolling `64px` downwards ⬇️ , the bottom navigation bar gets hidden. 
2. Upon scrolling `24px` upwards ⬆️  , the bottom navigation bar reappears.

File: src/components/BottomNav/BottomNav.tsx
Comment (rafalpawlow) [reply src/components/BottomNav/BottomNav.tsx]: This is a workaround to use hook inside render

**Review by drillprop — APPROVED** [review]
LGTM!

File: src/components/BottomNav/BottomNav.tsx
Comment (kdembler) [reply src/components/BottomNav/BottomNav.tsx]: I see now. However, I think this isn't a good idea. By "workaround" I guess you mean workaround for the eslint hooks rule. It's not there without a reason though. Rules of hooks say we cannot call hooks conditionally and that's what we're doing here by triggering inside a loop. You can create a wrapper for `NavLink` instead that will call the hook and manage it's `active` state autonomously

**Review by kdembler — COMMENTED** [review]
Looks good to me other than this one comment. However, let's also get an approval from @toiletgranny before merging

**Comment by toiletgranny** [comment]
Can we have the `scrollRange` set to `24px` on both scroll directions? Other than this, looks fine by me. :)

**Comment by kdembler** [comment]
After some more testing I've noticed that sometimes when I scroll to the top of the page, the navigation will disappear even though I haven't scrolled down. Here's an example video:

https://user-images.githubusercontent.com/12646744/135816841-ac6adb68-ae81-4972-bd67-844657210327.MP4


Are you able to replicate this? Maybe this is because of the scroll "bounce" off top?

**Comment by rafalpawlow** [comment]
> After some more testing I've noticed that sometimes when I scroll to the top of the page, the navigation will disappear even though I haven't scrolled down. Here's an example video:
> 
>  RPReplay_Final1633335141.MP4 
> Are you able to replicate this? Maybe this is because of the scroll "bounce" off top?

@kdembler Yeah I was able to replicate this on iOS in both directions. I made a fix for this.

**Review by kdembler — APPROVED** [review]
Great stuff, works really well!

**Pull request merged**
